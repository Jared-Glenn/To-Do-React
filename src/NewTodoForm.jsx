import React, { useState } from "react";

const NewTodoForm= (props) => {
    const INITIAL_STATE = {
        key: '',
        item: ''
    }
    const [formData, setFormData] = useState(INITIAL_STATE);
    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        props.addTodo({ ...formData });
        setFormData(INITIAL_STATE)
    }
    
    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="item">Item: </label>
            <input
                id="item"
                type="text"
                name="item"
                placeholder="Item Name"
                value={formData.item}
                onChange={handleChange}
            />
            <button>Add To Do Item</button>
        </form>
    )
}

export default NewTodoForm;