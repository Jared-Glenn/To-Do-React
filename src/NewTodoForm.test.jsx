import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

// Smoke Test
it('renders without crashing', () => {
    render(<NewTodoForm />);
});

// Snapshot Test
it('matches the snapshot', () => {
    const {asFragment} = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});