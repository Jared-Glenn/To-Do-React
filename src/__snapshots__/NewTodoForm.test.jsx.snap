// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="item"
    >
      Item: 
    </label>
    <input
      id="item"
      name="item"
      placeholder="Item Name"
      type="text"
      value=""
    />
    <button>
      Add To Do Item
    </button>
  </form>
</DocumentFragment>
`;
